sympy/sets/sets.py

```diff
@@ -1765,6 +1765,16 @@ class FiniteSet(Set, EvalfMixin):
     is_empty = False
     is_finite_set = True
 
+    def factor(self, *args, **kwargs):
+        """
+        Implements the SymPy factor routine, on the scalar parts
+        of a basis-dependent expression.
+        factor's documentation
+        ========================
+        """
+        fctr_components = [fctr(v, *args, **kwargs) * k for k, v in self.components.items()]
```

Comment:
I can't find any existing method named `fctr`.

Reply:
I just fixed this error and also trailing space one now some new error is coming :(

=>ImportError: cannot import name 'sqrt' from partially initialized module 'sympy'

Reply:
do i have make another PR for it?

Reply:
No, need to make a separate PR try importing `factor` inside the function as oscar has suggested.

Reply:
thanks so much :)